### ISSUE TITLE
Embedding api returns null (sometimes)

### ISSUE THREAD

**Issue by Gal-Lahat** [ollama/ollama#2140]
This is my code (C# .NET):

```cs
string url = "http://localhost:11434/api/embeddings";
string json = "{ \"model\": \"llama2:text\",\"prompt\": \"" + jsonSafeText + "\" }";

// get the response field from the json response
HttpClient client = new HttpClient();
var response = client.PostAsync(url, new StringContent(json, System.Text.Encoding.UTF8, "application/json")).Result;

if (response.StatusCode != System.Net.HttpStatusCode.OK)
{
Debug.LogError("Error getting embedding for: " + jsonSafeText);
return new float[0];
}

string responseString = response.Content.ReadAsStringAsync().Result;
```
On about 50% of the calls i get:
`{"embedding":null}` 
as response with no errors.

The issue persists on all models that I've tested (llama2, llama2:text, mistral, mistran:text)
 
The first run is always fine, but from the second run onwards it fail randomly with no error.

**Comment by alpe** [comment]
I was only able to replicate the issue on my box when the prompt is empty. For example:
```sh
curl -X POST http://localhost:11434/api/embeddings -d "{ \"model\": \"llama2\",\"prompt\": \"\" }"
```
Interestingly, the first call completes with the `{"embedding":null}` response but a second call freezes the instance. 🤷  This is a 🐛 . I can open a PR with a simple fix that rejects empty inputs. That should help.

I was running the server on OSX 14.3 with Apple M2.

**Comment by jmorganca** [comment]
This should be fixed now in 0.1.29. Sorry about this!
